language-plutus-core/plutus-ir/Language/PlutusIR/Parser.hs

```diff
@@ -120,7 +121,6 @@ inBraces = between lbrace rbrace
 
 reservedWords :: [T.Text]
 reservedWords =
-    map display PLC.allBuiltinNames ++
```

Comment:
Why remove the built-in names from the list of reserved words?  Because the compiler wraps builtins in lambdas with the same names and then if you print this out you can't re-parse it if the names are keywords. I removed keywords from the main parser completely a while ago but didn't want to start messing about with the PIR parser at that point. Once I've got this PR out of the way I'll have a look at the thing about putting type names into literal constants and when I'm doing that I may update the PIR parser as well.

Reply:
I see, thanks for the explanation.